src/components/uv-scroll.js

```diff
@@ -1,20 +1,79 @@
+const textureToData = new Map();
+const registeredTextures = [];
+
+export class UVScrollSystem {
+  tick(_t, dt) {
+    for (let i = 0; i < registeredTextures.length; i++) {
+      const map = registeredTextures[i];
+      const { offset, instances } = textureToData.get(map);
+      const { component } = instances[0];
+
+      offset.addScaledVector(component.data.speed, dt / 1000);
+
+      const increment = component.data.increment;
+      map.offset.x = increment.x ? offset.x - (offset.x % increment.x) : offset.x;
+      map.offset.y = increment.y ? offset.y - (offset.y % increment.y) : offset.y;
+
+      // TODO this should be handled by MobileStandardMaterial itself
```

Comment:
A bit confused why this is here -- is this the only thing that has ever changed a material's `uvTransform`? Why does `map.updateMatrix`only need to be called for `MobileStandardMaterial`?

Reply:
It's called by ThreeJS for the built in ThreeJS materials